### DISCUSSION TITLE
Move to using svg() to load blade-icons instead of blade icon components

### DISCUSSION THREAD

**Discussion by ianlandsman** [filamentphp/filament#3965] [General]
Using Blade Icons default component calls adds a huge performance hit when you have large icon libraries like Font Awesome. We've seen 200-400ms increases in page load time when using blade-icons with these large libraries. The Blade Icons docs don't mention this that I've seen but the config file does (https://github.com/blade-ui-kit/blade-icons/blob/1.x/config/blade-icons.php#L158)

Filament Notifications use the Blade Icons component syntax and so notifications break for anyone using the library with Blade Icons component syntax disabled. As Notifications don't have any particular need to use the component syntax as they are internal calls I suggest moving to the directive syntax (https://github.com/blade-ui-kit/blade-icons/blob/1.x/README.md?plain=1#L464) which has no performance penalties.

**Comment by danharrin** [comment]
Hey,

Just so you know, I have looked into it. Unfortunately, it would be a breaking change. This is because currently, Filament supports using any Blade component as an icon, not just those registered with Blade Icons. So by moving to use `@svg` everywhere, users would have their custom icons break and exceptions should be thrown.

So, we will need to wait for Filament v3 to be able to do this. That's not too far away, it will happen with Livewire v3 in a couple of months.

Right now, if you still want to use v2, you could publish the views using

```bash
php artisan vendor:publish --tag=notifications-views
php artisan vendor:publish --tag=filament-support-views
```

Then, search the published views for `<x-heroicon` and `<x-dynamic-component` and swap them out for the `@svg` helper.

Please make sure to delete any published views you dont change, after!

Dan

**Comment by ianlandsman** [comment]
Ah cool, thanks for the update!

**Comment by danharrin** [comment]
I've created an issue to track the development of this on our v3 roadmap - #4194

**Comment by danharrin** [comment]
The work for this is complete and it will be available in v3 - #4514
